File: marg/src/marg/token/Token.java
package marg.token;

public class Token {
	public Token(String string, TokenKind kind) {
		this.string = string;
		this.kind = kind;
	}
    public Token(char c, TokenKind kind) {
        string = String.valueOf(c);
        this.kind = kind;
    }
	
	public String string;
	public TokenKind kind;
	
	@Override
	public String toString() {
		return string + "\t: " + kind;
	}

    public boolean isEOF() {
        return kind == TokenKind.EOF;
    }
}

File: marg/src/marg/token/Token.scala
package marg.token


class Token private(string: String, kind: TokenKind) {

  override def toString = string.reverse.padTo(10, ' ').reverse.mkString + " : " + kind.toString
  def isEOF = kind eq TokenKind.EOF
  def String = string
  def Kind = kind
}

object Token {
  def apply(string: String, kind: TokenKind) = {
    require(string != null)
    new Token(string, kind)
  }
  def apply(char: Char, kind: TokenKind): Token = this(char.toString, kind)

  val EOF = Token("", TokenKind.EOF)
}

